homeassistant/helpers/config_validation.py

```diff
@@ -1662,10 +1664,34 @@ def STATE_CONDITION_SCHEMA(value: Any) -> dict[str, Any]:
     )
 )
 
+
+def _backward_compat_trigger_schema(value: Any | None) -> Any:
+    """Backward compatibility for trigger schemas."""
+
+    if not isinstance(value, dict):
+        return value
+
+    response = copy(value)
+
+    # `platform` has been renamed to `trigger`
+    if CONF_PLATFORM in value:
+        if CONF_TRIGGER in value:
+            raise vol.Invalid(
+                "Cannot specify both 'platform' and 'trigger'. Please use 'trigger' only."
+            )
```

Comment:
Wondering if this is still relevant since the `TRIGGER_BASE_SCHEMA` already makes them exclusive